gateway/modules/eventing/helpers.go

```diff
@@ -228,4 +242,16 @@ func (m *Module) selectRule(name string) (config.EventingRule, error) {
 		return rule, nil
 	}
 	return config.EventingRule{}, fmt.Errorf("could not find rule with name %s", name)
-}
\ No newline at end of file
+}
+
+func (m *Module) validate(event *model.QueueEventRequest) error {
+	if event.Type == utils.EventDBCreate || event.Type == utils.EventDBDelete || event.Type == utils.EventDBUpdate || event.Type == utils.EventFileCreate || event.Type == utils.EventFileDelete {
+		return nil
+	}
+	schema, p := m.schemas[event.Type]
+	if !p && m.config.IsForce == true {
+		return errors.New("required key not present in request")
+	}
+	_, err := m.schema.SchemaValidator(event.Type, schema, event.Payload.(map[string]interface{}))
```

Comment:
The logic is incorrect
- Return error if IsForce is true and key isn't present
- Return nil if key isn't present
- Validate schema if key is present